File: gherkin/java/src/main/java/gherkin/events/GherkinDocumentEvent.java
package gherkin.events;

import gherkin.ast.GherkinDocument;

public class GherkinDocumentEvent implements CucumberEvent {
    private final String type = "gherkin-document";
    public final String uri;
    public final GherkinDocument document;

    public GherkinDocumentEvent(String uri, GherkinDocument document) {
        this.uri = uri;
        this.document = document;
    }
}

File: gherkin/dotnet/Gherkin/Events/GherkinDocumentEvent.cs
using System.Runtime.Serialization;
using Gherkin.CucumberMessages.Types;

namespace Gherkin.Events
{
	public class GherkinDocumentEvent : IEvent
	{
	    public GherkinDocumentEvent(GherkinDocument eventArgs)
	    {
	        EventArgs = eventArgs;
	    }

	    public GherkinDocumentEvent()
	    {
	    }

	    [DataMember(Name = "gherkinDocument")]
	    public GherkinDocument EventArgs { get; set; }
	}
}